### ISSUE TITLE
fontscan: export list of system fonts

### ISSUE THREAD

**Issue by dominikh** [go-text/typesetting#141]
The `fontscan` package implements logic for enumerating the system fonts. However, this is currently only accessible via the FontMap and resolving faces based on rune, language, or family name. It would be useful if the raw list of system fonts were accessible, to be able to implement custom font fallback logic, provide lists of fonts for user configuration, etc.

The package does provide DefaultFontDirectories, but using this directly won't benefit from cached indices and requires additional I/O to get font metadata that is already available in `footprint`s.

**Comment by andydotxyz** [comment]
Oh, this is a good idea.
Fyne is just moving to the system font lookup too and this could be handy for some more advanced font configuration.

**Comment by benoitkugler** [comment]
What should we expose ? The 'footprint' struct and all its fields ? Have you some concrete use cases in mind ?

**Comment by dominikh** [comment]
> Have you some concrete use cases in mind ?

I mentioned some in my original comment. I myself want to implement my own font selection and fallback system and am interested in knowing about all available fonts (their locations, names, and attributes). Another common use case of knowing what fonts are available are font selection dropdowns, e.g. for configuring default fonts, or in desktop publishing software to select the font for a bit of text.

Basically, decouple font enumeration from font selection.

Exporting Footprint and its fields is probably a good start, yeah.
